src/sql/services/notebook/notebookService.ts

```diff
@@ -62,8 +72,18 @@ export interface INotebookManager {
 
 export interface INotebookParams extends IBootstrapParams {
 	notebookUri: URI;
+	input: NotebookInput;
 	providerId: string;
 	isTrusted: boolean;
 	profile?: IConnectionProfile;
 	modelFactory?: ModelFactory;
+}
+
+export interface INotebookEditor {
+	readonly notebookParams: INotebookParams;
+	readonly id: string;
```

Comment:
nice question. I think we need to handle rename scenarios? It's readonly though because you aren't expected to be able to change this property from the public API.